ipn/ipnlocal/cert.go

```diff
@@ -157,6 +162,44 @@ func shouldStartDomainRenewal(domain string, now time.Time, pair *TLSCertKeyPair
 	return false, nil
 }
 
+func (b *LocalBackend) shouldStartDomainRenewalARI(cs certStore, now time.Time, pair *TLSCertKeyPair) (bool, error) {
+	var blocks []*pem.Block
+	rest := pair.CertPEM
+	for len(rest) > 0 {
+		var block *pem.Block
+		block, rest = pem.Decode(rest)
+		if block == nil {
+			return false, fmt.Errorf("parsing certificate PEM")
+		}
+		blocks = append(blocks, block)
+	}
+	if len(blocks) < 2 {
+		return false, fmt.Errorf("could not parse certificate chain from certStore, got %d PEM block(s)", len(blocks))
+	}
+	ac, err := acmeClient(cs)
+	if err != nil {
+		return false, err
+	}
+	ctx, cancel := context.WithTimeout(b.ctx, 5*time.Second)
+	defer cancel()
+	d, err := ac.Discover(ctx)
+	if err != nil {
+		return false, err
+	}
+	ri, err := ac.FetchRenewalInfo(ctx, blocks[0].Bytes, blocks[len(blocks)-1].Bytes)
+	if err != nil {
+		return false, fmt.Errorf("failed to fetch renewal info from ACME server: %w", err)
+	}
+	fmt.Printf("==== ARI: %+v\n", ri)
```

Comment:
debug

Reply:
left it intentionally just to sanity-check that it's hitting the right code path :sweat_smile: 
will retest and remove this line after we get @noncombatant's change to x/crypto/acme cherry-picked into our fork